json_serializable/lib/src/check_dependencies.dart

```diff
@@ -37,21 +41,36 @@ Future<void> _validatePubspec(bool production, BuildStep buildStep) async {
     return;
   }
 
-  Future<Pubspec> readPubspec(AssetId asset) async {
-    final string = await buildStep.readAsString(asset);
-    return Pubspec.parse(string, sourceUrl: asset.uri);
+  final string = await buildStep.readAsString(pubspecAssetId);
+  final pubspec = Pubspec.parse(string, sourceUrl: pubspecAssetId.uri);
+
+  if (_checkAnnotationConstraint(pubspec, !production)
```

Comment:
Seemed the cleanest way to call the function and "do the thing" if the result is `null`.